### PR TITLE
Make vault client retry connection to overlay connection when handshake fails

### PR DESCRIPTION
**Pull request by b-yap**
Reconnection:

There are 2 threads that run independently (found in service.rs)
* `fn connection_handler()` - handles action from the connection
* `fn receiving_service()` - handles messages from the server
If connection fails, both should stop running.

In the case of an Error Message received, reconnection is required. 
example: **` Error{ code:ErrConf message:already-connected peer: `**:
* `fn connection_handler()` - relays the error message and return (which stops the thread)
* `fn receiving_service()` - no more data found in the read stream  and return (which stops the thread)

The error message is relayed inside `fn listen()` (found in agent.rs) and it goes through the "reconnect logic".

### CODE REVIEW

**Review by ebma — COMMENTED** [review]
How does it fix this? Wouldn't this be very worrying if our oracle agent is not able to get the proof all the time on mainnet? 🤔

**Comment by b-yap** [comment]
@ebma no, this is not yet final. I forgot to put it under _draft_. 

I'm testing the CI if "protocol 20" has perf adjustments. I'm planning to keep rerunning the test. Maybe we won't need to worry once mainnet moves to protocol 20 as well.

**Review by ebma — COMMENTED** [review]
I see you added some kind of prefix to most log messages. I'm fine with that and see how it could help. Let's maybe have it a single `:` instead of a `::` each time. This way it's more aligned to the default prefix of the log messages.

File: pallets/replace/src/lib.rs [hunk 1]
```diff
@@ -401,13 +401,16 @@ impl<T: Config> Pallet<T> {
...
-		// Emit RequestReplace event
-		Self::deposit_event(Event::<T>::RequestReplace {
+		let event = Event::<T>::RequestReplace {
 			old_vault_id: vault_id,
 			amount: to_be_replaced_increase.amount(),
 			asset: to_be_replaced_increase.currency(),
 			griefing_collateral: griefing_collateral.amount(),
-		});
+		};
+		log::info!("broadcasting event: {event:?}");
```
[the remark this answers is not in the document]
Comment (b-yap) [reply pallets/replace/src/lib.rs]: removed

File: clients/stellar-relay-lib/src/connection/services.rs [hunk 2]
```diff
@@ -252,7 +252,7 @@ async fn _connection_handler(
 		// start the connection to Stellar node with a 'hello'
 		ConnectorActions::SendHello => {
 			let msg = connector.create_hello_message(time_now())?;
-			log::trace!("Sending Hello Message...{:?}", msg);
+			log::info!("_connection_handler():: Sending Hello Message...{:?}", msg);
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/stellar-relay-lib/src/connection/services.rs]: updated

File: clients/stellar-relay-lib/src/connection/overlay_connection.rs [hunk 3]
```diff
@@ -57,41 +57,42 @@ impl StellarOverlayConnection {
 		let res = self.relay_message_receiver.recv().await;
 
 		// Reconnection only when the maximum number of retries has not been reached.
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/stellar-relay-lib/src/connection/overlay_connection.rs:50]: removed

File: clients/stellar-relay-lib/src/connection/services.rs [hunk 4]
```diff
@@ -286,42 +286,41 @@ pub(crate) async fn connection_handler(
...
+			Ok(None) => {
+				log::warn!("connection_handler():: Unexpected empty response from receiver");
+			},
+
+			Err(_) => {
+				log_error!(
+					connector.send_to_user(StellarRelayMessage::Timeout).await,
+					format!("connection_handler():: Connection Timed out")
+				);
+				return
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/stellar-relay-lib/src/connection/services.rs]: not really. In the previous code, it returns after the max number of retries is exhausted. 

I've placed the "retry" logic back, similar to `fn receiving_service()` here in services.rs

**Pull request merged**